src/main/java/org/openstreetmap/atlas/geography/atlas/raw/slicing/RawAtlasSlicer.java

```diff
@@ -1,258 +1,767 @@
+                countries.addAll(
+                        Arrays.asList(getCountryBoundaryMap().getCountryCodeISO3(slice.last())
+                                .getIso3CountryCode().split(ISOCountryTag.COUNTRY_DELIMITER)));
+                tags.put(ISOCountryTag.KEY,
+                        String.join(ISOCountryTag.COUNTRY_DELIMITER, countries));
+                tags.put(SyntheticBoundaryNodeTag.KEY, SyntheticBoundaryNodeTag.YES.toString());
+                final CompletePoint syntheticBoundaryNode = new CompletePoint(1L, slice.last(),
+                        tags, new HashSet<>());
+                syntheticBoundaryNode.withIdentifier(
+                        pointIdentifierGenerator.generateIdentifier(syntheticBoundaryNode));
+                this.stagedPoints.put(syntheticBoundaryNode.getIdentifier(), syntheticBoundaryNode);
+            }
+            else
+            {
+                this.stagedPoints.get(pointsAtLastLocation.iterator().next().getIdentifier())
+                        .withAddedTag(SyntheticBoundaryNodeTag.KEY,
+                                SyntheticBoundaryNodeTag.EXISTING.toString());
+            }
         }
-        throw new CoreException(
-                "All raw Atlas lines must have a country code by the time Relation slicing is done. One of the two Lines {} or {} does not!",
-                one.getIdentifier(), two.getIdentifier());
     }
 
-    public RawAtlasSlicer(final AtlasLoadingOption loadingOption,
-            final CoordinateToNewPointMapping newPointCoordinates, final Atlas startingAtlas)
+    /**
+     * Given a remainder geometry for a sliced multipolygon Relation, add in new Lines to the Atlas
+     *
+     * @param newRelation
+     *            A new sliced Relation
+     * @param remainder
+     *            Geometry not covered by this new Relation's existing sliced Line members
+     * @param role
+     *            The role for this geometry in the Relation
+     * @return A Set of ids for the newly added Lines, used to updated the
+     *         SyntheticRelationMemberAdded tag in the new Relation
+     */
+    private Set<String> addSyntheticLinesForRemainder(final CompleteRelation newRelation,
+            final Geometry remainder, final String role)
     {
-        this.loadingOption = loadingOption;
-        this.newPointCoordinates = newPointCoordinates;
-        this.startingAtlas = startingAtlas;
-        this.shardOrAtlasName = getShardOrAtlasName(startingAtlas);
+        final Optional<String> countryCodeTag = newRelation.getTag(ISOCountryTag.KEY);
+        if (countryCodeTag.isEmpty())
+        {
+            throw new CoreException("Could not find country code for split relation {}",
+                    newRelation.getIdentifier());
+        }
+        final String countryCode = countryCodeTag.get();
+        final Set<String> synetheticIdsAdded = new HashSet<>();
+        for (int i = 0; i < remainder.getNumGeometries(); i++)
```

Comment:
One of the rare places where Checkstyle allows less than 3 characters in a variable!